src/app/core/basic-datatypes/entity/entity.datatype.ts

```diff
@@ -46,19 +47,59 @@ export class EntityDatatype extends StringDatatype {
     super();
   }
 
-  override importMapFunction(
+  /**
+   * Maps a value from an import to an actual entity in the database by comparing the value with the given field of entities.
+   * Handles type conversion between numbers and strings to improve matching.
+   *
+   * @param val The value from an import that should be mapped to an entity reference.
+   * @param schemaField The config defining details of the field that will hold the entity reference after mapping.
+   * @param additional The field of the referenced entity that should be compared with the val.
+   * @returns Promise resolving to the ID of the matched entity or undefined if no match is found.
+   */
+  override async importMapFunction(
     val: any,
     schemaField: EntitySchemaField,
-    additional?: any,
-  ) {
-    if (!additional) {
-      return Promise.resolve(undefined);
+    additional?: string,
+  ): Promise<string | undefined> {
+    if (!additional || val == null) {
+      return undefined;
+    }
+
+    const normalizedVal = this.normalizeValue(val);
+
+    try {
+      const entities = await this.entityMapper.loadType(schemaField.additional);
+      const matchedEntity = entities.find((entity) => {
+        const entityFieldValue = this.normalizeValue(entity[additional]);
+        return entityFieldValue === normalizedVal;
+      });
+
+      return matchedEntity?.getId();
+    } catch (error) {
+      Logging.error("Error in EntityDatatype importMapFunction:", error);
+      return undefined;
     }
-    return this.entityMapper
-      .loadType(schemaField.additional)
-      .then((res) => res.find((e) => e[additional] === val)?.getId());
   }
 
+  /**
+   * Normalizes a value for comparison, converting it to a standardized string format.
+   * Ensures both numbers and strings are treated consistently.
+   *
+   * @param val The value to normalize.
+   * @returns The normalized value as a string.
+   */
+  private normalizeValue(val: any): string {
```

Comment:
```suggestion
  function normalizeValue(val: any): string {
```

This doesn't need anything from the class, so we can make it a simple function and above, when using it, write `normalizeValue()` instead of `this.normalizeValue()` (which is even shorter).